src/utils/getAbsoluteHooksPath.js

```diff
@@ -0,0 +1,40 @@
+// @flow
+import execa from 'execa'
+import path from 'path'
+
+const getAbsoluteHooksPath = async (hookName: string) => {
+  let absoluteHooksPath
+
+  // First retrieve the hooks path from the git config
+  // It can be either absolute or relative according to git documentation
+  // NOTE: this requires git >= 2.9 (2016-06-13)
+  const hooksPathCommand = await execa('git', [
+    'config',
+    '--get',
+    'core.hooksPath'
+  ])
+  const hooksPath = hooksPathCommand.stdout || '.git/hooks'
+
+  if (path.isAbsolute(hooksPath)) {
+    // If the hooks path is absolute, return it
+    absoluteHooksPath = hooksPathCommand.stdout
+  } else {
+    // If it is relative, retrieve the absolute path of the git directory then map the relative hooks path to it
+    // NOTE: this requires git >= 2.13 (2017-05-10)
+    const absoluteGitDirCommand = await execa('git', [
+      'rev-parse',
+      '--absolute-git-dir'
+    ])
+
+    // Finally, reassemble those paths
+    absoluteHooksPath = path.resolve(
+      // Use dirname to remove the trailing "/.git" from the path
+      path.dirname(absoluteGitDirCommand.stdout),
+      hooksPath
+    )
+  }
+
+  return path.resolve(absoluteHooksPath, hookName)
+}
+
+export default getAbsoluteHooksPath
```

Comment:
I don't understand why we need to check if it's absolute or not, can we make something like this?

```suggestion
// @flow
import execa from 'execa'
import path from 'path'

const getGitHooksPath = async (hookName: string): string => {
  const gitCoreHooksPath = await execa('git', [
    'config',
    '--get',
    'core.hooksPath'
  ])

  const defaultHooksPath = await execa('git', [
    'rev-parse',
    '--absolute-git-dir'
  ]) + '/hooks'
  
  const hooksPath = gitCoreHooksPath.stdout || defaultHooksPath.stdout

  return path.resolve(hooksPath, hookName)
}

export default getGitHooksPath
```

Reply:
Somehow from your previous review, I thought you wanted to always ensure working with an absolute path, but I guess using just a relative path as you illustrate it works too :)

We would get the following results, based on the value of `core.hooksPath`

- `/etc/git/hooks` --> `/etc/git/hooks/prepare-commit-msg`
- `undefined` --> `/path/to/project/.git/hooks/prepare-commit-msg`
- `.git/hooks` --> `.git/hooks/prepare-commit-msg`
- `.husky` --> `.husky/prepare-commit-msg`

When my proposal would instead always return an absolute path:

- `/etc/git/hooks` --> `/etc/git/hooks/prepare-commit-msg`
- `undefined` --> `/path/to/project/.git/hooks/prepare-commit-msg`
- `.git/hooks` --> `/path/to/project/.git/hooks/prepare-commit-msg`
- `.husky` --> `/path/to/project/.husky/prepare-commit-msg`